scripts/changelog/changelog.go

```diff
@@ -0,0 +1,176 @@
+package main
+
+import (
+    "bufio"
+    "fmt"
+    "os"
+    "strings"
+    "time"
+)
+
+type Change struct {
+    Version string
+    Name string
+    Datetime time.Time
+    Changes []string
+}
+
+const (
+    CHANGE_TEMPLATE = "./CHANGE_TEMPLATE"
+    TOP_LEVEL = "../../CHANGELOG.md"
+    AMAZON_LINUX = "../../packaging/amazon-linux-ami/ecs-init.spec"
+    SUSE = "../../packaging/suse/amazon-ecs-init.spec"
+    UBUNTU = "../../packaging/ubuntu-trusty/debian/changelog"
+
+    AMAZON_LINUX_TIME_FMT = "Mon Jan 02 2006"
+    DEBIAN_TIME_FMT = "Mon, 02 Jan 2006 15:04:05 -0700"
+    SUSE_TIME_FMT = "Mon Jan 02, 15:04:05 MST 2006"
```

Comment:
Does this get used anywhere? We don't need to add changelog for SUSE?

Reply:
We will add changelog for SUSE when we update the SUSE rpm process.  I can remove this SUSE-specific line if necessary.  Originally I was planning on doing all three changelogs at once, but the SUSE packaging is very out of date and will need some work.